proto/proto/dex.proto

```diff
@@ -21,33 +23,44 @@ message Swap {
 // signature, as it is only capable of consuming the NFT from a
 // Swap transaction.
 message SwapClaim {
-  // Proves the spent note records 1 unit of the swap NFT.
+  // Contains the SwapClaim proof.
   bytes zkproof = 1;
+  // Encapsulates the authorized fields of the SwapClaim action, used in signing.
+  SwapClaimBody body = 2;
+}
+
+// Encapsulates the authorized fields of the SwapClaim action, used in signing.
+message SwapClaimBody {
   // The nullifier for the Swap NFT to be consumed.
-  crypto.Nullifier nullifier = 2;
+  crypto.Nullifier nullifier = 1;
   // The fee allows `SwapClaim` without an additional `Spend`.
-  transaction.Fee fee = 3;
+  transaction.Fee fee = 2;
   // Note output for asset 1.
-  crypto.NotePayload output_1 = 4;
+  crypto.NotePayload output_1 = 3;
   // Note output for asset 2.
-  crypto.NotePayload output_2 = 5;
+  crypto.NotePayload output_2 = 4;
   // Block root for the associated `Swap` action, identifying the block the `Swap` was included in.
-  crypto.MerkleRoot anchor = 6;
-  // Price for asset 1 in the associated `Swap`.
-  uint64 price_1 = 7;
-  // Price for asset 2 in the associated `Swap`.
-  uint64 price_2 = 8;
+  crypto.MerkleRoot anchor = 5;
+  // Input and output amounts for the assets in the swap.
+  BatchSwapOutputData output_data = 6;
+  // The trading pair to swap.
+  TradingPair trading_pair = 7;
 }
 
 // The authorized data of a Swap transaction.
 message SwapBody {
   // The trading pair to swap.
   TradingPair trading_pair = 1;
-  // Commitment to the amount for asset 1 (delta 1).
-  bytes asset_1_commitment = 2;
-  // Commitment to the amount for asset 2 (delta 2).
-  bytes asset_2_commitment = 3;
-  // A commitment to the fee's contribution to the transaction's value balance.
+  // @exclude These will become commitments when flow encryption/ABCI++ are available
+  // The amount for asset 1.
+  uint64 delta_1 = 2;
+  // The amount for asset 2.
+  uint64 delta_2 = 3;
+  // @exclude // Commitment to the amount for asset 1 (delta 1).
+  // @exclude bytes delta_1_commitment = 2;
+  // @exclude // Commitment to the amount for asset 2 (delta 2).
+  // @exclude bytes delta_2_commitment = 3;
```

Comment:
what does `@exclude` do?

Reply:
It is a keyword in the protobuf doc generator that excludes the comment from the docs